app/controllers/place_controller.rb

```diff
@@ -0,0 +1,93 @@
+require "slimmer/headers"
+
+class PlaceController < ApplicationController
+  before_filter :redirect_if_api_request
+  before_filter -> { set_expiry unless viewing_draft_content? }
+
+  INVALID_POSTCODE_ERROR = "invalidPostcodeError".freeze
+  NO_LOCATION_ERROR = "validPostcodeNoLocation".freeze
+
+  CUSTOM_SLUGS = {
```

Comment:
@whoojemaflip Have a look at https://github.com/alphagov/frontend/pull/1070/commits/09aa005b29376fccedbe3620993a3a323215cdf6. Initially we moved everything into a separate controller method but found there was quite a bit of duplication in the controller and controller tests, as the only way the 'report-child-abuse' differs is in the templates that it renders. What do you think?